File: src/renderer/src/types/export.ts
export interface ExportFreeSSHResponse {
  data: any
  filename: string
}

export interface ExportOpenSSHResponse {
  data: string
  filename: string
}

export interface ImportFreeSSHRequest {
  data: Uint8Array
}

export interface ImportFreeSSHResponse {
  connections_imported: number
  groups_imported: number
  port_forwards_imported: number
  errors?: string[]
}

export interface ImportOpenSSHRequest {
  data: Uint8Array
}

export interface ImportOpenSSHResponse {
  connections_imported: number
  keys_imported: number
  errors?: string[]
}

File: backend/internal/models/export.go
package models

type ExportRequest struct {
	Format string `json:"format"` // "freessh"
}

type ExportResponse struct {
	Data     []byte `json:"data"`
	Filename string `json:"filename"`
}

type ImportRequest struct {
	Format string `json:"format"` // "freessh"
	Data   []byte `json:"data"`
}

type ImportResponse struct {
	ConnectionsImported  int      `json:"connections_imported"`
	GroupsImported       int      `json:"groups_imported"`
	PortForwardsImported int      `json:"port_forwards_imported"`
	Errors               []string `json:"errors,omitempty"`
}
